### ISSUE TITLE
testing new models

### ISSUE THREAD

**Issue by micsanMercy16** [facebookresearch/fairseq-lua#48]
Hi Jonas, just a curiosity. What are the correct formats and steps to test new models in other languages, please? Please advise. Thank you.

**Comment by michaelauli** [comment]
Do you mean how to *train* and test new models? The README has example steps on how to do that. If you want to train a larger model, then check the READMEs of the pre-trained models which detail the training settings for those models.

**Comment by micsanMercy16** [comment]
Hi Michael, thank you. I meant to ask if corpora must be in .sh format in order to be prepared to be trained. Thank you.

**Comment by michaelauli** [comment]
What do you mean by .sh format?

**Comment by micsanMercy16** [comment]
I have some data I would like to train with fairseq and README says cd data, bash prepare-iwslt14.sh. So in order to train my own data could I just apply same command bash prepare-data.sh? Thank you.

**Comment by micsanMercy16** [comment]
This is what I get when I try to train my data, please advise. Thank you.
fairseq micheles$ fairseq train -sourcelang es -targetlang en -datadir data-bin/span-samples-sources.tokenized.es-en -model fconv -nenclayer 4 -nlayer 3 -dropout 0.2  -optim nag -lr 0.25 -clip 0.1 -momentum 0.99 -timeavg -bptt 0 -savedir trainings/fconv
/Users/micheles/torch/install/bin/luajit: ...es/torch/install/share/lua/5.1/fairseq/scripts/train.lua:29: assertion failed!
stack traceback:
	[C]: in function 'assert'
	...es/torch/install/share/lua/5.1/fairseq/scripts/train.lua:29: in main chunk
	[C]: in function 'require'
	...install/lib/luarocks/rocks/fairseq-cpu/scm-1/bin/fairseq:17: in main chunk
	[C]: at 0x010513dbc0

**Comment by jgehring** [comment]
@micsanMercy16 you'll need a machine with a NVidia GPU for training new models. The error message stems from [train.lua, line 29](https://github.com/facebookresearch/fairseq/blob/master/train.lua#L29). As you can see, the line above states that cuda is required for training.

**Comment by micsanMercy16** [comment]
Does it not require a -cpu instead?

**Comment by michaelauli** [comment]
The current training code does not support CPU training. You can, however,
generate with already trained model on CPU.
